wire-moshi-support/src/main/java/com/squareup/wire/ByteStringJsonAdapter.java

```diff
@@ -0,0 +1,31 @@
+package com.squareup.wire;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.JsonReader;
+import com.squareup.moshi.JsonWriter;
+import java.io.IOException;
+import javax.annotation.Nullable;
+import okio.ByteString;
+
+import static com.squareup.moshi.JsonReader.Token;
+
+class ByteStringJsonAdapter extends JsonAdapter<ByteString> {
```

Comment:
if you declare this as a single instance you can call `.nullSafe()` to not need the null handling

Reply:
doing as what we're doing in `wire-gson-support` for now